rs/boundary_node/salt_sharing/canister/canister.rs

```diff
@@ -0,0 +1,161 @@
+use std::cell::RefCell;
+use std::time::Duration;
+
+use crate::storage::{LAST_SALT_NS, SALT, SALT_SIZE};
+use crate::time::delay_till_next_month;
+use candid::Principal;
+use ic_cdk::{api::time, spawn};
+use ic_cdk_macros::{init, post_upgrade, query};
+use ic_cdk_timers::set_timer;
+use rand_chacha::{
+    rand_core::{RngCore, SeedableRng},
+    ChaCha20Rng,
+};
+use salt_api::{GetSaltError, GetSaltResponse, InitArg, SaltGenerationStrategy, SaltResponse};
+
+// A small delay after beginning of the month when salt is generated
+const START_OF_MONTH_DELAY: Duration = Duration::from_secs(5);
+// Maximum number of attempts to initialize random seed
+const MAX_SEED_INIT_ATTEMPTS: i32 = 15;
```

Comment:
why i32 for a _number_?

Reply:
Well `i32` doesn't look like a bad idea for a number. `u32` is slightly better, did You imply that?

Reply:
Indeed. I'd expect an unsigned here for a counter.